sygnal/gcmpushkin.py

```diff
@@ -347,16 +347,24 @@ async def _dispatch_notification_unlimited(
         with self.sygnal.tracer.start_span(
             "gcm_dispatch", tags=span_tags, child_of=context.opentracing_span
         ) as span_parent:
+            # TODO: Implement collapse_key to queue only one message per room.
+            failed: List[str] = []
+
             data = GcmPushkin._build_data(n, device)
+
+            # Reject pushkey if default_payload is misconfigured
+            if not data:
```

Comment:
We will hit this branch if `data is None` (good), but also if `data` is an empty dict (maybe bad?) because empty dictionaries are falsey:

```python
>>> bool({})
False
```

Is this intended? If so, the error message is confusing.

(Prefer `x is None` or `x is not None` when testing for none-ness)

Reply:
I should have tested for None, have fixed.